evap/staff/importers.py

```diff
@@ -97,28 +98,98 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
+            )
+
+    def check_existing_course(self, semester, importer):
+        name_collision = self.check_name_collision(semester)
+        if name_collision:
+            # Only if there are name collisions, it is worth to check for an existing course
+            if self.find_and_set_existing_course(semester, importer):
+                return True
+            if name_collision == "both":
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} ({}) does already exist in this semester.").format(self.name_en, self.name_de)
+                )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _("Course {} does already exist in this semester.").format(
+                        self.name_en if name_collision == "name_en" else self.name_de
+                    )
+                )
+        return False
+
+    def check_name_collision(self, semester):
+        name_collision = None
+        if Course.objects.filter(semester=semester, name_en=self.name_en).exists():
+            name_collision = "name_en"
+        if Course.objects.filter(semester=semester, name_de=self.name_de).exists():
+            name_collision = "name_de" if not name_collision else "both"
+        return name_collision
+
+    def find_and_set_existing_course(self, semester, importer):
+        courses = Course.objects.filter(
             semester=semester,
+            name_en=self.name_en,
+            name_de=self.name_de,
         )
-        course.save()
-        course.responsibles.set([responsible_dbobj])
-        course.degrees.set(self.degrees)
-        evaluation = Evaluation(
-            vote_start_datetime=vote_start_datetime,
-            vote_end_date=vote_end_date,
-            course=course,
-            wait_for_grade_upload_before_publishing=self.is_graded,
-        )
-        evaluation.save()
-        evaluation.contributions.create(
-            evaluation=evaluation,
-            contributor=responsible_dbobj,
-            role=Contribution.Role.EDITOR,
-            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-        )
+        if courses.exists():
+            course = courses.first()
+            differing_attributes = []
+            if course.type != self.course_type:
+                differing_attributes.append("course type")
+            if len(course.responsibles.all()) != 1 or course.responsibles.first().email != self.responsible_email:
+                differing_attributes.append("responsible person")
+            if not differing_attributes:
+                evaluations = Evaluation.objects.filter(course=course)
+                if len(evaluations) == 1:
+                    if evaluations.first().wait_for_grade_upload_before_publishing == self.is_graded:
+                        self.existing_course = course
+                        return course
```

Comment:
I think this should probably only do one of those two things, I would only set the instance var which would go along nicely with the suggested change to `check_existing_course`

Reply:
Yeah I should read all comments before replying to them one by one :D
I totally agree ^^